src/value_generators/test/DocumentGeneratorTestCases.yml

```diff
@@ -177,6 +183,19 @@ Tests:
           xYYXCh/Ybl: { "$numberLong": "9" }
           EHcmXZX/d4: { "$numberLong": "10" }
 
+  - Name: Object Generator disallow duplicate keys
+    GivenTemplate:
+      a:
+        {^Object: {
+          withNEntries: 3,
+          havingKeys: "c",
+          andValues: {^Inc: {}},
+          allowDuplicateKeys: false
+        }}
+    ThenReturns:
+      - a:
+          c: { "$numberLong": "1" }
```

Comment:
YAML doesn't support duplicated mappings. I don't understand how these tests work and how to check that when the duplicates are allowed, the do make it through into the object. The test above with random strings happens to generate unique keys.

Reply:
Ya this is un-testable in the current yaml-based testing mechanism. I think it's worth adding a comment to the test that the test-case of allowing duplicate keys is not tested just so someone in the future knows to take extra caution if they mess with this part of the code.